publisher/internal/analytics/ethereum/checks.go

```diff
@@ -12,10 +12,21 @@ func isUniswapPositionsNFT(address string) bool {
 	return false
 }
 
-func isStableOrNativeInvolved(position Position) bool {
+func isStableInvolved(position Position) bool {
```

Comment:
This definitely could be a `Position` method. Perhaps called `hasStableCoin` or even `hasStable`.
The same idea applies to `isNativeInvolved` => `hasNative`.

However, seeing how these functions are essentially the same, it would be even better to refactor them into one method called `hasToken(token ...string)` and simply call it with appropriate parameters where needed(the signature may also accept a slice of strings - up to you).